### DISCUSSION TITLE
Error: HTTP 404 fetching XXXX.fasta.fai when using --load=symlink

### DISCUSSION THREAD

**Discussion by lznbb** [GMOD/jbrowse-components#2778] [Q&A] [answered]
Hi,
I was running jbrowse2 with a Linux web version. It worked well when I was using
**jbrowse add-assembly [fasta]--load=copy --target=[path]**
However, when I tried to set --load=symlink while no any other change with the command, the assembly cannot show. From the error message, it seems cannot recognize the fasta index file. I tried to remove the symbolic link for the index file and copy it manually. It does work. But I'm wondering if there is a better solution.
Thanks

**Comment by cmdcolin** [comment, marked as answer]
Hi there
Can you post what version of the CLI you are using? I tested out with v1.6.5 and it did work as far as I could tell

This could be related to https://github.com/GMOD/jbrowse-components/issues/2685 which was just fixed v1.6.5


You can upgrade by re-running

npm install -g @jbrowse/cli

**Reply by lznbb** [reply]
hmmm... that's weird. i ran jbrowse -v and got **@jbrowse/cli/1.6.5 linux-x64 node-v14.18.2**, but still it's not working

**Reply by cmdcolin** [reply]
can you post what is in the config.json file and where the symlinks point to (e.g. ll -l of the directory)

**Reply by cmdcolin** [reply]
might try to make sure to clear out old broken symlinks if they exist before re-running command

**Reply by lznbb** [reply]
{
  "assemblies": [
    {
      "name": "AP1815",
      "sequence": {
        "type": "ReferenceSequenceTrack",
        "trackId": "AP1815-ReferenceSequenceTrack",
        "adapter": {
          "type": "IndexedFastaAdapter",
          "fastaLocation": {
            "uri": "AP1815.fasta",
            "locationType": "UriLocation"
          },
          "faiLocation": {
            "uri": "AP1815.fasta.fai",
            "locationType": "UriLocation"
          }
        }
      }
    }
  ],
  "configuration": {},
  "connections": [],
  "defaultSession": {
    "name": "New Session"
  },
  "tracks": []
}
that's my config.json for your information. Each time I run the command, I would run **rm -rf** previously to remove the old symlinks and config files.

**Reply by cmdcolin** [reply]
can you check if the symlink is valid? you can run "file /path/to/the/symlink" and it will output "broken symbolic link" if broken or "symbolic link" if it works. if it works, then it might be that your webserver is not following symlinks, in which case you might need to look at your webserver's configuration e.g. https://superuser.com/questions/244245/how-do-i-get-apache-to-follow-symlinks

**Reply by lznbb** [reply]
thanks for the suggestion. i followed it and saw the symbolic link is valid. actually i'm not using an apache webserver and not sure if the issue is related to that...

**Reply by lznbb** [reply]
to my understanding, if the issue is that symlinks not enabled, there should be problems with both .fasta and .fasta.fai file. but in this case, the fasta file works well while fasta index file cannot be found.

**Reply by cmdcolin** [reply]
jbrowse has to fetch the fasta index file first to tell which parts of the fasta file to download, it might be that both dont work but that the index file is just the first error message you see

**Reply by cmdcolin** [reply]
jbrowse itself has no server side code that is interpreting symlinks after the config file is created so it will likely come down to whatever server you use, if you let us know can try to help troubleshoot :)

**Reply by lznbb** [reply]
i said only the index file has problem because it worked well after replacing the index file symbolic link by copying the real file, while the fasta file just remained symlink. for web server, i think it is just booted by **npx serve .** and i did not find any other one based on my docker image. that's why i think it's really strange...

**Reply by cmdcolin** [reply]
The serve command I believe actually requires a special argument to handle symlinks, see https://github.com/vercel/serve-handler#options  and npx serve --help (I think you can use npx serve . --symlinks) to help

**Reply by lznbb** [reply]
this time it worked perfectly with me. Thank you very much for your patience!
